cpp/cmake/CUDF_Thirdparty.cmake

```diff
@@ -0,0 +1,189 @@
+include(FetchContent)
+
+###################################################################################################
+# - thrust/cub ------------------------------------------------------------------------------------
+
+message("Fetching Thrust")
+
+CPMFindPackage(
+  NAME Thrust
+  GITHUB_REPOSITORY NVIDIA/thrust
+  GIT_TAG 1.10.0
+  VERSION 1.10.0
+  GIT_SHALLOW TRUE
+  OPTIONS
+    # If there is no pre-installed thrust we can use, we'll install our fetched copy
+    # together with CUDF
+    "THRUST_INSTALL TRUE"
+  )
+
+thrust_create_target(cudf::Thrust FROM_OPTIONS)
+
+###################################################################################################
+# - find arrow ------------------------------------------------------------------------------------
+
+if(ARROW_STATIC_LIB)
+  message(STATUS "BUILDING ARROW")
+  CPMFindPackage(
+    NAME Arrow
+    GITHUB_REPOSITORY google/googletest
+    GIT_TAG apache-arrow-1.0.1
+    VERSION 1.0.1
+    GIT_SHALLOW TRUE
+    OPTIONS
+      "ARROW_WITH_LZ4 OFF"
+      "ARROW_WITH_ZSTD OFF"
+      "ARROW_WITH_BROTLI OFF"
+      "ARROW_WITH_SNAPPY OFF"
+      "ARROW_WITH_ZLIB OFF"
+      "ARROW_BUILD_STATIC ON"
+      "ARROW_BUILD_SHARED OFF"
+      "ARROW_BOOST_USE_SHARED OFF"
+      "ARROW_BUILD_TESTS OFF"
+      "ARROW_TEST_MEMCHECK OFF"
+      "ARROW_BUILD_BENCHMARKS OFF"
+      "ARROW_IPC ON"
+      "ARROW_FLIGHT OFF"
+      "ARROW_COMPUTE OFF"
+      "ARROW_CUDA ON"
+      "ARROW_JEMALLOC OFF"
+      "ARROW_BOOST_VENDORED OFF"
+      "ARROW_PYTHON OFF"
+      "ARROW_USE_GLOG OFF"
+      "ARROW_DATASET ON"
+      "ARROW_BUILD_UTILITIES OFF"
+      "ARROW_HDFS OFF"
+      "CMAKE_VERBOSE_MAKEFILE ON"
+  )
```

Comment:
If someone sets `ARROW_STATIC_LIB=ON` and has Arrow installed it will be found as opposed to built.

Reply:
That `GITHUB_REPOSITORY` up there looks like a copy&paste error ;)